python/dgl/distributed/graph_services.py

```diff
@@ -625,28 +660,58 @@ def sample_etype_neighbors(
         nodes = F.cat(homo_nids, 0)
 
     def issue_remote_req(node_ids):
+        if prob is not None:
+            # See NOTE 1
+            _prob = [
+                g.edges[etype].data[prob].kvstore_key
+                if prob in g.edges[etype].data
+                else ""
+                for etype in g.etypes
+            ]
+        else:
+            _prob = None
         return SamplingRequestEtype(
             node_ids,
-            etype_field,
             fanout,
             edge_dir=edge_dir,
-            prob=prob,
+            prob=_prob,
             replace=replace,
             etype_sorted=etype_sorted,
         )
 
     def local_access(local_g, partition_book, local_nids):
-        return _sample_etype_neighbors(
-            local_g,
-            partition_book,
-            local_nids,
-            etype_field,
-            fanout,
-            edge_dir,
-            prob,
-            replace,
-            etype_sorted=etype_sorted,
-        )
+        etype_offset = gpb._local_etype_offset
+        # See NOTE 1
+        if prob is None:
+            return _sample_etype_neighbors(
+                local_g,
+                partition_book,
+                local_nids,
+                etype_offset,
+                fanout,
+                edge_dir,
+                None,
+                replace,
+                etype_sorted=etype_sorted,
+            )
+        else:
+            _prob = [
+                g.edges[etype].data[prob].local_partition
+                if prob in g.edges[etype].data
+                else None
+                for etype in g.etypes
+            ]
+            return _sample_etype_neighbors(
+                local_g,
+                partition_book,
+                local_nids,
+                etype_offset,
+                fanout,
+                edge_dir,
+                _prob,
+                replace,
+                etype_sorted=etype_sorted,
+            )
```

Comment:
Can we merge L686-L696 and L704-L714 to reduce code duplication?

Reply:
Yes.